backend/db/refresh.py

```diff
@@ -37,19 +37,28 @@ def refresh_db(
     this is set to True."""
     print('INFO: Starting database refresh.', flush=True)  # flush: for gh action
     t0, t0_str = datetime.now(), current_datetime()
-    with get_db_connection(local=use_local_database) as con:
-        last_refresh = last_refresh_timestamp(con)
-        if since and dp.parse(since) > dp.parse(last_refresh) and not force_non_contiguity:
-            raise ValueError(SINCE_ERR)
-        since = since if since else last_refresh
+    update_db_status_var('last_refresh_request', t0_str)
+    update_db_status_var('refresh_status', 'active')
+    try:
+        with get_db_connection(local=use_local_database) as con:
+            last_refresh = last_refresh_timestamp(con)
+            if since and dp.parse(since) > dp.parse(last_refresh) and not force_non_contiguity:
+                raise ValueError(SINCE_ERR)
+            since = since if since else last_refresh
 
-        # Refresh DB
-        csets_and_members_enclave_to_db(con, schema, since)
+            # Refresh DB:
+            csets_and_members_enclave_to_db(con, schema, since)
```

Comment:
@hugwuoke This was the only bug I found! Needed to be intended under the `with` block in order for the connection to be open when running that function. In these `with` blocks database connections or file connections will close after the block completes.